components/filter/filter.js

```diff
@@ -0,0 +1,253 @@
+ * @fires d2l-filter-change - Dispatched when a dimension's value(s) have changed
+ */
+class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			/**
+			 * Disables the dropdown opener for the filter
+			 */
+			disabled: { type: String, reflect: true },
+			_activeDimensionKey: { type: String },
+			_dimensions : { type: Array }
+		};
+	}
+
+	static get styles() {
+		return [bodyStandardStyles, css`
+			div[slot="header"] {
+				padding: 18px 6px;
+			}
+			.d2l-filter-dimension-header {
+				align-items: center;
+				display: flex;
+				width: 100%;
+			}
+			.d2l-filter-dimension-header-text {
+				padding-right: calc(2rem + 2px);
+				text-align: center;
+				width: 100%;
+			}
+			:host([dir="rtl"]) .d2l-filter-dimension-header-text {
+				padding-left: calc(2rem + 2px);
+				padding-right: 0;
+			}
+
+			.d2l-filter-dimension-set-value-text {
+				overflow: hidden;
+				text-overflow: ellipsis;
+				white-space: nowrap;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this._dimensions = [];
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		// Prevent these events from bubbling out of the filter
+		this.addEventListener('d2l-hierarchical-view-hide-complete', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-hide-start', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-show-complete', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-show-start', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-resize', this._stopPropagation);
+	}
+
+	render() {
+		const header = this._buildHeader();
+		const dimensions = this._buildDimensions();
+
+		return html`
+			<d2l-dropdown-button-subtle
+				@d2l-dropdown-close="${this._handleDropdownClose}"
+				@d2l-dropdown-open="${this._stopPropagation}"
+				@d2l-dropdown-position="${this._stopPropagation}"
+				text="${this.localize('components.filter.filters')}"
+				?disabled="${this.disabled}">
+				<d2l-dropdown-menu min-width="285" max-width="420" no-padding-header>
+					${header}
+					<d2l-menu label="${this.localize('components.filter.filters')}">
+						${dimensions}
+					</d2l-menu>
+				</d2l-dropdown-menu>
+			</d2l-dropdown-button-subtle>
+			<slot
+				@d2l-filter-dimension-data-change="${this._handleDimensionDataChange}"
+				@slotchange="${this._handleSlotChange}">
+			</slot>
+		`;
+	}
+
+	_buildDimension(dimension) {
+		let dimensionHTML;
+		switch (dimension.type) {
+			case 'd2l-filter-dimension-set':
+				dimensionHTML = this._createSetDimension(dimension);
+				break;
+		}
+		return html`
+			<d2l-hierarchical-view
+				@d2l-hierarchical-view-show-complete="${this._handleDimensionShowComplete}"
+				@d2l-hierarchical-view-show-start="${this._handleDimensionShowStart}"
+				data-key="${dimension.key}">
```

Comment:
I previously used `id` here, but in the HM filter framework guids are used as the keys for a dimension, and an `id` can't start with a number.  Is this a good use case for data attributes instead?  I figured this was cleaner than still using `id` and adding `dim-` to the front of it, and then striping that off all the places below where I use this value to grab dimension data (places can be seen in https://github.com/BrightspaceUI/core/pull/1301/commits/96c9cb5054fc0090ff6315dd2726ad677ed7c49e)

Reply:
I think so, yeah. Like `id`s have all kinds of other restrictions on them (uniqueness, format, etc.) and the browser ends up treating them specially for CSS/creating hashes for quick lookup... so if we don't actually _need_ to reference the thing by id, no need to use one.